filebeat/beater/filebeat.go

```diff
@@ -102,6 +103,9 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	// Blocks progressing. As soon as channel is closed, all defer statements come into play
 	<-fb.done
 
+	// Uses shutdown_timeout to wait for potential completion of sending events
+	<-time.After(fb.config.ShutdownTimeout)
```

Comment:
Original problem with shutdown timeout was: wait for enqueued events to be ACKed on shutdown. This requires the crawler/prospectors being closed before doing the timeout. In addition, we really always want to wait, or rather wait for max timeout on shutdown until all events are ACKed -> continue killing filebeat once events have been killed.

Introducing a timer here won't achieve either, as this only delays processing of the kill signal.